---
layout: tutorial_page
permalink: /bicg_2018_recap
title: BiCG Recap
header1: Bioinformatics for Cancer Genomics 2018
header2: So how do I process my data?
image: /site_images/CBW_cancerDNA_icon-16.jpg
home: https://bioinformaticsdotca.github.io/bicg_2018
---

# Analysis workflows

As the modules mentioned, there are multiple workflows that have been developed to help you process your data. The Broad Institute has developed a set of [best practices](https://software.broadinstitute.org/gatk/best-practices/workflow) when doing certain analysis as part of it's Genome Analysis Toolkit. Flowcharts for their pipelines are available at their website, and cover:
* [Data pre-processing](https://software.broadinstitute.org/gatk/best-practices/workflow?id=11165)
* [Detecting somatic SNVs/Indels](https://software.broadinstitute.org/gatk/best-practices/workflow?id=11146)
* [Detecting germline SNVs/Indels](https://software.broadinstitute.org/gatk/best-practices/workflow?id=11145)
* [Detecting somatic CNVs](https://software.broadinstitute.org/gatk/best-practices/workflow?id=11147)

In addition to the workflows determined by the Broad institute, a few other workflows can be found here:

* *RNA-Sequencing pipeline*
<img src="https://github.com/bioinformaticsdotca/BiCG_2018/blob/master/Workflows/Images/RNA_workflow_diffexp.jpg?raw=true" alt="Differential Expression" width="850" />



